extensions/ipynb/src/cellAttachmentRenderer.ts

```diff
@@ -0,0 +1,41 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type * as MarkdownIt from 'markdown-it';
+import type * as MarkdownItToken from 'markdown-it/lib/token';
+import type { RendererContext } from 'vscode-notebook-renderer';
+
+interface MarkdownItRenderer {
+	extendMarkdownIt(fn: (md: MarkdownIt) => void): void;
+}
+
+export async function activate(ctx: RendererContext<void>) {
+	const markdownItRenderer = (await ctx.getRenderer('vscode.markdown-it-renderer')) as MarkdownItRenderer | any;
+	if (!markdownItRenderer) {
+		throw new Error(`Could not load 'vscode.markdown-it-renderer'`);
+	}
+
+	markdownItRenderer.extendMarkdownIt((md: MarkdownIt) => {
+		const original = md.renderer.rules.image;
+		md.renderer.rules.image = (tokens: MarkdownItToken[], idx: number, options, env, self) => {
+			const token = tokens[idx];
+			const src = token.attrGet('src');
+			const attachments: Record<string, Record<string, string>> = (env.outputItem.metadata as any).custom.attachments;
```

Comment:
You may need to check that `env.outputItem.metadata` and `env.outputItem.metadata.custom` are not undefined before accessing properties on them

Reply:
Newest commit should address this, I think that 
`attachments = env.outputItem.metadata?.custom?.attachments` + `if (attachments && src)` 
should handle it?